source/fundamentals/monitoring.txt

```diff
@@ -0,0 +1,406 @@
+- Registers these MXBean instances with the platform MBean server. 
+
+MXBeans registered on the platform MBean server have the following properties:
+
+.. list-table::
+   :header-rows: 1
+   :stub-columns: 1
+
+   * - Property
+     - Description
+
+   * - ``clusterId``
+     - A client-generated unique identifier. This identifier ensures that
+       each MXBean the driver makes has a unique name when an application has 
+       multiple ``MongoClient`` instances connected to the same MongoDB deployment.
+
+   * - ``host``
+     - The hostname of the machine running the ``mongod`` or ``mongos`` process.
+
+   * - ``port``
+     - The port on which the ``mongod`` or ``mongos`` process is listening.
+
+   * - ``minSize``
+     - The minimum size of the connection pool, including idle and in-use connections.
+
+   * - ``maxSize``
+     - The maximum size of the connection pool, including idle and in-use connections.
+
+   * - ``size``
+     - The current size of the connection pool, including idle and in-use connections.
+
+   * - ``checkedOutCount``
+     - The current count of connections that are in use.
+
+
+All MXBean instances created by the driver are under the domain
+``"org.mongodb.driver"``.
+
+For more information on the topics discussed in this subsection, see the
+following resources:
+
+- `Oracle Platform MBean Server Reference documentation <https://docs.oracle.com/en/java/javase/16/management/overview-java-se-monitoring-and-management.html#GUID-F7B9AB8A-F5A8-472A-AEC6-93B5B7FBE7CE>`__
+- `Oracle MXBean documentation <https://docs.oracle.com/javase/tutorial/jmx/mbeans/mxbeans.html>`__
+
+JMX and JConsole Example
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+This example shows how you can monitor the driver's connection pools using JMX
+and **JConsole**. JConsole is a JMX compliant GUI monitoring tool that comes with
+the Java Platform.
+
+The following code snippet adds a ``JMXConnectionPoolListener`` to a
+``MongoClient`` instance. The code then starts a JMX connection server exposing
+the platform MBean server on port 9999 for JConsole to connect to.
+
+.. literalinclude:: /includes/fundamentals/code-snippets/JMXMonitoring.java
+   :language: java
+   :dedent:
+   :start-after: start jmx-example
+   :end-before: end jmx-example
+
+The above code snippet should produce output that looks like this:
+
+.. code-block:: none
+   :copyable: false
+
+   Start the connector server...
+   The connector server started.
+   Press <Enter> to stop the server.
+
+Once you have started your server, open JConsole in your terminal using the
+following command:
+
+.. code-block:: shell
+   
+   jconsole
+
+.. note:: Consult the Official JConsole Documentation
+
+   The descriptions of the JConsole GUI on this page are illustrative
+   rather than a source of truth. For guaranteed up to date information, consult 
+   `the official JConsole documentation <https://www.oracle.com/technical-resources/articles/java/jconsole.html>`__. 
```

Comment:
Issue: Let's show confidence in our documentation. Instead, you can omit this or put this after your steps (at line 395) to say:

```suggestion
For more information on starting up and using JConsole, visit `the official JConsole documentation <https://www.oracle.com/technical-resources/articles/java/jconsole.html>`__. 
```

Reply:
As JConsole is a third party tool that may change without us knowing, I believe it is appropriate to provide a disclaimer that readers should expect discrepancies in our guide and their experience.

In a previous review documenting 3rd party technologies, [a disclaimer similar to this was requested by Jeff Yemin](https://github.com/mongodb/docs-java/pull/76#discussion_r623846300)